scripts/Diffraction/isis_powder/hrpd.py

```diff
@@ -60,6 +62,48 @@ def mask_prompt_pulses_if_necessary(self, ws_list):
         for ws in ws_list:
             self._mask_prompt_pulses(ws)
 
+    def should_subtract_empty_inst(self):
+        return self._inst_settings.subtract_empty_inst
+
+    def create_solid_angle_corrections(self, vanadium, run_details):
+        """
+        Creates the solid angle corrections from a vanadium run, only applicable on HRPD otherwise return None
+        :param vanadium: The vanadium used to create this
+        :param run_details: the run details of to use
+        """
+        if not self._inst_settings.do_solid_angle:
+            return None
```

Comment:
I don't think this necessarily needs to be changed, but something to keep in mind: `return` returns None implicitly, and is generally preferred when using a return statement in the same way as a break. Using `return` over `return None` makes it clear to something looking at this method that the returned value is not actually caught and used anywhere.